src/bin/rustfmt.rs

```diff
@@ -138,6 +152,32 @@ fn determine_operation<I>(opts: &Options, args: I) -> Operation
         return Operation::Help;
     }
 
+    // if no file argument is supplied, read from stdin
+    if matches.free.len() != 1 {
+
+        // make sure the write mode is plain or not set
+        // (the other options would require a file)
+        match matches.opt_str("write-mode") {
+            Some(mode) => {
+                match mode.parse() {
+                    Ok(WriteMode::Plain) => (),
+                    _ => return Operation::InvalidInput("Using stdin requires write-mode to be \
+                                                         plain"
+                                                            .into()),
+                }
+            }
+            _ => (),
+        }
```

Comment:
I was looking at how `gofmt` does it
```
echo "1+1" | gofmt -w                                                                                                                                                         
> error: cannot use -w with standard input
```
hence I decided to give an error message. It has upsides and downsides. It's more graceful but might surprise some users.

But I actually agree with you, the only thing I would do then is add a warning in the `--help` usage notes that stdin will always print using plain.